cvat/apps/engine/views.py

```diff
@@ -616,10 +614,32 @@ def data(self, request, pk):
                     status=status.HTTP_400_BAD_REQUEST)
             serializer = DataSerializer(data=request.data)
             serializer.is_valid(raise_exception=True)
+            data = {k: v for k, v in serializer.validated_data.items()}
+
+            tmp_path = os.path.join('./data/tmp', pk)
+            os.makedirs(tmp_path, exist_ok=True)
+            if data.get('use_chunk_upload'): # write chunk
+                with open(os.path.join(tmp_path, data.get('chunk_file_name')), 'ab+') as destination:
+                    destination.write(data.get('client_files')[0]['file'].read())
+            else: # write each file
+                for client_file in data.get('client_files'):
+                    with open(os.path.join(tmp_path, client_file['file'].name), 'ab+') as destination:
+                        destination.write(client_file['file'].read())
+            if 'end_of_upload' in request.data and not data.get('end_of_upload'):
+                return Response(data='Chunk is delivered', status=status.HTTP_202_ACCEPTED)
```

Comment:
Do you need this response text ``Chunk is delivered``? I think status with code is enough

Reply:
Removed